src/hypothesis/searchstrategy/types.py

```diff
@@ -46,6 +48,12 @@ def try_issubclass(thing, maybe_superclass):
         return False
 
 
+def deferred_resolve(thing):
```

Comment:
Is there a reason to not make *all* calls to from\_type return a deferred? I'd strongly prefer that if there's not. deferred is pretty low-overhead, and it ensures that all calls to from\_type are correct by construction and we don't have to remember to call deferred\_resolve whenever we might introduce recursion.